stetho/src/main/java/com/facebook/stetho/dumpapp/plugins/CrashDumperPlugin.java

```diff
@@ -0,0 +1,173 @@
+    out.println("    <code>: Exit code (default: " + OPTION_EXIT_DEFAULT + ")");
+  }
+
+  private void doSystemExit(Iterator<String> argsIter) {
+    String exitCodeStr = ArgsHelper.nextOptionalArg(argsIter, OPTION_EXIT_DEFAULT);
+    System.exit(Integer.parseInt(exitCodeStr));
+  }
+
+  private void doKill(DumperContext dumpContext, Iterator<String> argsIter) throws DumpException {
+    String signal = ArgsHelper.nextOptionalArg(argsIter, OPTION_KILL_DEFAULT);
+    try {
+      Process kill = new ProcessBuilder()
+          .command("/system/bin/kill", "-" + signal, String.valueOf(android.os.Process.myPid()))
+          .redirectErrorStream(true)
+          .start();
+
+      // Handle kill command output gracefully in the event that the signal delivered didn't
+      // actually take out our process...
+      try {
+        InputStream in = kill.getInputStream();
+        Util.copy(in, dumpContext.getStdout(), new byte[1024]);
+      } finally {
+        kill.destroy();
+      }
+    } catch (IOException e) {
+      throw new DumpException("Failed to invoke kill: " + e);
+    }
+  }
+
+  private void doUncaughtException(Iterator<String> argsIter) throws DumpException {
+    String throwableClass = ArgsHelper.nextOptionalArg(argsIter, OPTION_THROW_DEFAULT);
+    try {
+      Thread crashThread = new Thread(
+          new ThrowRunnable(
+              (Class<? extends Throwable>)Class.forName(throwableClass)));
+      crashThread.start();
+
+      CountDownLatch impossibleLatch = new CountDownLatch(1);
+      Util.awaitUninterruptibly(impossibleLatch);
+    } catch (NoSuchMethodException | ClassNotFoundException | ClassCastException e) {
+      throw new DumpException("Invalid supplied Throwable class: " + e);
+    }
+  }
+
+  private static class ThrowRunnable implements Runnable {
+    @Nullable
+    private final Constructor<? extends Throwable> mDetailMessageConstructor;
+
+    private final Constructor<? extends Throwable> mParameterlessConstructor;
+
+    public ThrowRunnable(Class<? extends Throwable> throwableClass) throws NoSuchMethodException {
+      mDetailMessageConstructor = tryGetDeclaredConstructor(throwableClass, String.class);
+      mParameterlessConstructor = throwableClass.getDeclaredConstructor();
+    }
+
+    @Nullable
+    private static <T> Constructor<? extends T> tryGetDeclaredConstructor(
+        Class<T> clazz,
+        Class<?>... parameterTypes) {
+      try {
+        return clazz.getDeclaredConstructor(parameterTypes);
+      } catch (NoSuchMethodException e) {
+        return null;
+      }
+    }
+
+    @Override
+    public void run() {
+      try {
+        Throwable t = createThrowable();
+        ExceptionUtil.<Error>sneakyThrow(t);
+      } catch (InstantiationException | IllegalAccessException | InvocationTargetException e) {
+        // Hehe, ok whatever :)
+        ExceptionUtil.<Error>sneakyThrow(e);
```

Comment:
Ack'd, this design was admittedly sloppy.  I'll tidy it up so that it's much clearer that the throw is the only thing which must be on a separate thread.